salt/addons/init.sls

```diff
@@ -14,40 +10,44 @@ include:
     - source:      salt://addons/addons/namespace.yaml.jinja
     - template:    jinja
 
-{% if salt['pillar.get']('addons:dns', 'false').lower() == 'true' %}
-/etc/kubernetes/addons/kubedns-sa.yaml:
-  file.managed:
-    - source:     salt://addons/addons/kubedns-sa.yaml.jinja
-    - template:   jinja
-
-/etc/kubernetes/addons/kubedns-cm.yaml:
-  file.managed:
-    - source:      salt://addons/addons/kubedns-cm.yaml.jinja
-    - template:    jinja
+apply-namespace:
+  cmd.run:
+    - name: |
+        kubectl apply -f /etc/kubernetes/addons/namespace.yaml
```

Comment:
Reading https://github.com/kubernetes/kubernetes/issues/44165, it seems that this is an issue iff the kubernetes manifests that are being run by the two "kubectl apply" calls are different. In that case, it's possible that the 'later one' isn't actually applied. In this case however, the two manifests are both the same so I think we should be okay.

Regardless, I agree with Kiall that it would be useful to have something that runs "once per cluster". I also agree that we should look at implementing something like that separately since it's out of the scope of this Pull Request.